### PR TITLE
HIPP-1222: Update domain and sub-domain values on API details page

### PR DESCRIPTION
**Pull request by PaulCDurham**
https://jira.tools.tax.service.gov.uk/browse/HIPP-1222

### CODE REVIEW

File: app/config/Domains.scala [hunk 1]
```diff
@@ -17,23 +17,77 @@
...
+  def getDomainDescription(apiDetail: ApiDetail): Option[String] = {
+    getDomainDescription(apiDetail.domain)
+  }
+
+  def getSubDomain(domainCode: String, subDomainCode: String): Option[SubDomain] = {
+    getDomain(domainCode)
+      .flatMap(
+        domain =>
+          domain.subDomains
+            .find(_.code.toLowerCase.equals(subDomainCode.trim.toLowerCase))
```
Comment (rdawson-hmrc) [inline app/config/Domains.scala:53]: ditto

File: app/config/Domains.scala [hunk 2]
```diff
@@ -17,23 +17,77 @@
...
 
 import scala.jdk.CollectionConverters.CollectionHasAsScala
 
+trait Domains {
+
+  def domains: Seq[Domain]
+
+  def getDomain(domainCode: String): Option[Domain] = {
+    domains
+      .find(_.code.toLowerCase.equals(domainCode.trim.toLowerCase))
```
[the remark this answers is not in the document]
Comment (rdawson-hmrc) [reply app/config/Domains.scala:31]: maybe a 'normalise' function applied to both codes would be a good idea, could use it for subdomains as well?

[the remark this answers is not in the document]
Comment (PaulCDurham) [reply app/config/Domains.scala]: All codes are now normalised (trimmed and converted to lower case) before comparisons. Added some unit tests for that.

File: app/config/Domains.scala (hunk 1 above)
Comment (PaulCDurham) [reply app/config/Domains.scala]: Fixed, ditto

**Review by rdawson-hmrc — APPROVED** [review]

**Pull request merged**